### ISSUE TITLE
No battle, blocking or mine damage

### ISSUE THREAD

**Issue by Piwoslaw** [freeorion/freeorion#2140]
### Environment

* **FreeOrion Version**: FreeOrion_2018-06-04.41c7665_Test_Win32
* **Operating System**: Windows 10, 64-bit
* **Graphic card used**:
* **Fetched as** 
Weekly development build

[save-20180608_065759.zip](https://github.com/freeorion/freeorion/files/2083008/save-20180608_065759.zip)

![fo08062018](https://user-images.githubusercontent.com/19825312/41140590-a7b469e2-6aee-11e8-987e-f53f5f31eb8b.jpg)

### Description
- No battle between my and enemy fleets, even when both parked in the same system for multiple turns.
- My fleet does not damage even scouts nor troop ships.
- Enemy fleet can pass right through systems where my fleet is stationed.
- My fleet in a system does not block enemy supply, enemy fleet does not block my supply.
- No damage to enemy ships in my system with mine tech.

### Steps to reproduce
Previous game (no save files) was OK, but new game has above symptoms. I tried to save, close FO, open and load game, but bug persists.

**Comment by Vezzra** [comment]
@Piwoslaw, I just checked your saved game, and the explanation for your problem is very simple: you are at peace with all other empires. Combat, mine damage etc. only happens if you are at war.

So no bug at all here, everything works just as intended. 😉 

Closing this as invalid.

**Comment by Piwoslaw** [comment]
Hi Vezzra,I checked and yes - I was at peace with everyone. After declaring war the AIs now have no chance:)I was not aware that this is on by default? In the previous game I played yesterday there was not issue with peace/war.Anyhow, thank you for the pointer and sorry for troubling you.Cheers,Piwoslaw

**Comment by Vezzra** [comment]
> I was not aware that this is on by default?

It's not, or it shouldn't. Are you sure you didn't propose peace to them? Because if you didn't, that would mean it's a bug.

So, if you can replicate this situation (being at peace with the other empires despite never having proposed peace to them), please open another issue.

**Comment by Piwoslaw** [comment]
I didn't propose peace because I didn't even know how until now (I never use the 'Empires' window). If it can be done some other way, then maybe I clicked something by accident, I can't rule that out. Or maybe... When FO initiates a new game, then the Empires window is open by default? If yes, then possibly I clicked on it while it was loading, though I see that just a click is not enough, it would have to be a right-click, then another.If it does happen again, and I can confirm that I didn't initiate it myself, then I will collect as much info as possible and open a new report.

**Comment by Vezzra** [comment]
> If it can be done some other way, then maybe I clicked something by accident, I can't rule that out.

AFAIK, that can only be done via the Empires window, and it's *highly* unlikely that you accidentally proposed peace to *two* empires, with both of them accepting.

> When FO initiates a new game, then the Empires window is open by default?

Yes.

> If yes, then possibly I clicked on it while it was loading, though I see that just a click is not enough, it would have to be a right-click, then another.

And that must have happened twice, see above. I think we can rule that out.

> If it does happen again, and I can confirm that I didn't initiate it myself, then I will collect as much info as possible and open a new report.

Keep an eye on it, especially at the beginning of your games. You should always see a reddish icon with two crossed swords, not the yellowish peace symbol left of the empire names. Most importantly, we need some way to reliably reproduce the issue, so you need to pay attention to what you did in order for that issue to happen.

**Comment by Piwoslaw** [comment]
Hi Vezzra,I played another game and all was well. Today I started a new game and noticed that an enemy ship did not attack my scout. So I check the Empires window and again - both enemy empires had a peace icon. I have attached the earliest autosave from that game.I then generated 6-8 new games ('Quick Start' on the Home page) to see what happens. It appears that when the new game is generated, all empires are at peace with me by default, but only after 2-3 seconds the peace icon changes to the crossed swords.Could it be that if I click on something during those first 2-3 seconds the status does not change? Or maybe some other process on the computer would disrupt the change to battle? Though I admit that I do not remember if I started clicking that soon after the new game generated.A possible solution may be to have the empires at war by default when a new game starts, not at peace and then changed to war?Should I open a new bug report?

**Comment by Dilvish-fo** [comment]
I have found how to reproduce this issue-- it is caused in recent-ish versions when loading a turn 1 autosave as the very first autosave of the game it is done even before any player actions are taken (whereas all other autosaves, even turn-stat autosaves, will include AI orders for that turn), and there is a newish setting under which when a game is loaded the AI by default does not make any new orders for that turn.

>  It appears that when the new game is generated, all empires are at peace with me by default, but only after 2-3 seconds the peace icon changes to the crossed swords.Could it be that if I click on something during those first 2-3 seconds the status does not change? Or maybe some other process on the computer would disrupt the change to battle? 

The time delay is because the AI clients have to be started and parse all the content, and then the AI's each declare war on everyone else in the game.  

It would be good to fix this for 0.4.8.

One possible solution would be to have the default status be War as you suggest.

Although the c++ clients have access to the client Orders, I do not see any way exposed to python where the AI could check after a game load if it has any orders for the current turn,  but something just to check for them being nonempty would be fairly easy to add, and in that case the AI could go ahead and play out its full turn.  I think that would be my long term preferred solution.

A simple but clunky hack, which I would not want to put into master but which I confirmed works and might be OK for 0.4.8 is simply to have that AI setting not apply to turn 1 saves.  @Morlic-fo  opinions?

**Comment by Piwoslaw** [comment]
Yup, I can confirm that in this game I went back and used the autosave from turn 1.

**Comment by Dilvish-fo** [comment]
> I've pushed some untested code to expose the OrderSet to the AI interface.

Thanks, I do still think that can wind up useful, although not as simply as I had hoped.  [As noted in the related 'hack/fix' PR](https://github.com/freeorion/freeorion/pull/2156#issuecomment-397800560), it turns out that even this 'turn 0' autosave, which is described in the options as being before any gameplay, still winds up including the AI's first turn orders:

>> Also, with the galaxy generation save, Production and Colonization orders are correctly stored in the savegame with the AI.

>Yes, I double checked mine now, and that is the case for me also for the turn 0 autosave, and examining the code that seems to make tenative sense to me.  After galaxy creation, with this gamestart save option the Server server requests the client save state info and goes into WaitingForSaveData state, in this state it defers handling of all Diplomatic messages, so even if the AIs generate their war declarations before returning their orders and savestate, handling of those declarations would get deferred until after the server finished saving the game.

I am thinking that perhaps in ```HumanClientFSM.cpp  boost::statechart::result WaitingForGameStart::react(const GameStart& msg)```, when the galaxy creation save is being initiated, the FSM should pause a few seconds before moving on, so that the server would have time to query the AIs for their savestate before the HumanClient sends out the PlayingTurn message.  Then checking a turn 1 loadstate for the absence of orders would work the way I'd hoped.   I'd also need to doublecheck the AIClient FSM to make sure it could respond properly at that point.  Anyways, that's just a thought, I need to look at the AIClient FSM more to figure out what could work, or if it might need some adjustments also.

**Comment by Vezzra** [comment]
> One possible solution would be to have the default status be War as you suggest.

Wouldn't that be a very simple and straightforward fix? Also, it could and probably should go into master, as having "peace" as the default diplomatic status isn't really a good idea as long as the AI isn't capable of basic diplomacy anyway. Then we could remove that "AIs declare war on everyone at game start" fix (which I consider to be a hack anyway).

I'd very much prefer this to any release branch only hacks or more complicated fixes, at this point. Less potential for the "introducing new bugs when fixing an existing one" thing...
